### PR TITLE
Meeting scheduling: save initialized schedule in xlsx file

### PR DESCRIPTION
**Pull request by ge0ffrey**
Needed to diagnose a bug in the Parametrization work branch

### CODE REVIEW

File: optaplanner-examples/src/main/java/org/optaplanner/examples/meetingscheduling/persistence/MeetingSchedulingXlsxFileIO.java [hunk 1]
```diff
@@ -317,6 +329,45 @@ private void readMeetingDuration(Meeting meeting) {
                     .collect(toList());
         }
 
+        private TimeGrain extractTimeGrain(Meeting meeting, Map<LocalDateTime, TimeGrain> timeGrainMap) {
+            String dateString = nextStringCell().getStringCellValue();
+            String startTimeString = nextStringCell().getStringCellValue();
+            if (!dateString.isEmpty() || !startTimeString.isEmpty()) {
```
Comment (rsynek) [inline optaplanner-examples/src/main/java/org/optaplanner/examples/meetingscheduling/persistence/MeetingSchedulingXlsxFileIO.java]: As both dateString and startTimeString are used in this branch, is the logical OR correct here?

File: optaplanner-examples/src/main/java/org/optaplanner/examples/meetingscheduling/persistence/MeetingSchedulingXlsxFileIO.java [hunk 2]
```diff
@@ -317,6 +329,45 @@ private void readMeetingDuration(Meeting meeting) {
...
+
+                TimeGrain timeGrain = timeGrainMap.get(dateTime);
+                if (timeGrain == null) {
+                    throw new IllegalStateException(currentPosition() + ": The meeting with id (" + meeting.getId()
+                            + ") has a timeGrain date (" + dateString + ") and startTime (" + startTimeString
+                            + ") that doesn't exist in the other sheet (Day).");
+                }
+                return timeGrain;
+            }
+            return null;
```
Comment (rsynek) [inline optaplanner-examples/src/main/java/org/optaplanner/examples/meetingscheduling/persistence/MeetingSchedulingXlsxFileIO.java]: soft suggestion: returning null on one place might result in NPE in other place. `Optional.empty()` could be used in such situations.

File: optaplanner-examples/src/main/java/org/optaplanner/examples/meetingscheduling/persistence/MeetingSchedulingXlsxFileIO.java [hunk 3]
```diff
@@ -486,6 +542,15 @@ private void writeMeetings() {
                         meeting.getPreferredAttendanceList().stream()
                                 .map(preferredAttendance -> preferredAttendance.getPerson().getFullName())
                                 .collect(joining(", ")));
+                List<MeetingAssignment> meetingAssignmentList = meetingAssignmentMap.get(meeting);
+                if (meetingAssignmentList.size() != 1) {
+                    throw new IllegalStateException("Impossible state");
```
Comment (rsynek) [inline optaplanner-examples/src/main/java/org/optaplanner/examples/meetingscheduling/persistence/MeetingSchedulingXlsxFileIO.java]: another soft suggestions: error message could be more descriptive about what impossible state happened.

File: optaplanner-examples/src/main/java/org/optaplanner/examples/meetingscheduling/persistence/MeetingSchedulingXlsxFileIO.java (hunk 1 above)
Comment (ge0ffrey) [reply optaplanner-examples/src/main/java/org/optaplanner/examples/meetingscheduling/persistence/MeetingSchedulingXlsxFileIO.java]: Yes, so it fails fast if only one them is filled in.

File: optaplanner-examples/src/main/java/org/optaplanner/examples/meetingscheduling/persistence/MeetingSchedulingXlsxFileIO.java (hunk 2 above)
Comment (ge0ffrey) [reply optaplanner-examples/src/main/java/org/optaplanner/examples/meetingscheduling/persistence/MeetingSchedulingXlsxFileIO.java]: if the field is empty, the timegrain of the MeetingAssignment should be null

File: optaplanner-examples/src/main/java/org/optaplanner/examples/meetingscheduling/persistence/MeetingSchedulingXlsxFileIO.java (hunk 3 above)
Comment (ge0ffrey) [reply optaplanner-examples/src/main/java/org/optaplanner/examples/meetingscheduling/persistence/MeetingSchedulingXlsxFileIO.java]: Agreed, will fix in Parameterization branch

**Pull request merged**